presto-native-execution/presto_cpp/main/PrestoServer.cpp

```diff
@@ -1441,16 +1441,20 @@ void PrestoServer::checkOverload() {
         : 0;
     const bool isMemOverloaded =
         (currentUsedMemoryBytes > overloadedThresholdMemBytes);
-    if (isMemOverloaded) {
-      LOG(WARNING) << "Server memory is overloaded. Currently used: "
+    if (isMemOverloaded && !isMemOverloaded_) {
+      LOG(WARNING) << "OVERLOAD: Server memory is overloaded. Currently used: "
                    << velox::succinctBytes(currentUsedMemoryBytes)
+                   << ", including tracked: "
+                   << velox::succinctBytes(pool_->usedBytes())
                    << ", threshold: "
                    << velox::succinctBytes(overloadedThresholdMemBytes);
-    } else if (isMemOverloaded_) {
-      LOG(INFO) << "Server memory is no longer overloaded. Currently used: "
-                << velox::succinctBytes(currentUsedMemoryBytes)
-                << ", threshold: "
-                << velox::succinctBytes(overloadedThresholdMemBytes);
+    } else if (!isMemOverloaded && isMemOverloaded_) {
+      LOG(INFO)
+          << "OVERLOAD: Server memory is no longer overloaded. Currently used: "
+          << velox::succinctBytes(currentUsedMemoryBytes)
+          << ", including tracked: " << velox::succinctBytes(pool_->usedBytes())
+          << ", threshold: "
+          << velox::succinctBytes(overloadedThresholdMemBytes);
     }
     RECORD_METRIC_VALUE(kCounterOverloadedMem, isMemOverloaded ? 100 : 0);
```

Comment:
Let's just have a flag to indicate if it is overloaded or not, and we don't have to report as a pct.

Reply:
As we discussed offline, we try to use counter to see if works for Meta internal infra.